genvm/core/types.go

```diff
@@ -100,6 +102,7 @@ type Host interface {
 	Handler() Handler
 	Template() Template
 	Layer() LayerID
+	GetGenesisID() Hash20
```

Comment:
how about an interface `Hash([]byte)` instead and the template gets to decide how to hash?

Reply:
i prefer current version

Reply:
my rationale is that in the future custom templates don't necessarily care about genesis id, as long as the template can parse its own tx correctly.

Reply:
it is up to template how to implement replay protection, they are not necessarily will use hashing for that